### PR TITLE
Add FSPoll backend

### PR DESCRIPTION
**Pull request by princemaple**

### CODE REVIEW

File: lib/file_system/backends/fs_poll.ex [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  defp files_mtimes(dirs, files_mtimes_map \\ %{}) do
+    Enum.reduce(dirs, files_mtimes_map, fn dir, map ->
+      case File.stat!(dir) do
+        %{type: :regular, mtime: mtime} ->
+          Map.put(map, dir, mtime)
+        %{type: :directory} ->
+          dir
+          |> Path.join("**/*")
+          |> Path.wildcard
+          |> files_mtimes(map)
```
Comment (falood) [inline lib/file_system/backends/fs_poll.ex]: `Path.wildcard("#{base}/**/*")` will process sub dirs, and `files_mtimes` will do `wildcard` again, so file maybe processed multiple times.

**Review by falood — COMMENTED** [review]
Looks good and thank you very much!
but I think we'd better have unittest for this feature 🙂

File: lib/file_system/backends/fs_poll.ex (hunk 1 above)
Comment (princemaple) [reply lib/file_system/backends/fs_poll.ex]: Right!

**Comment by princemaple** [comment]
Added a test for diff. Let me know if there are other things you'd like to test.

**Comment by princemaple** [comment]
Bump for another look?

**Comment by falood** [comment]
worry for the delay, merged.
Thank you very much!

**Comment by princemaple** [comment]
👍

**Pull request merged**